file_catalog/schema/validation.py

```diff
@@ -88,57 +89,69 @@ def _find_all_field_vals(metadata: types.Metadata, fields: List[str]) -> Dict[st
         return field_vals
 
     @staticmethod
-    def _field_vals_are_different(field: str, val: Any, old_metadata: types.Metadata) -> bool:
+    def _field_vals_are_different(
+        field: str, val: Any, old_metadata: types.Metadata
+    ) -> bool:
         """Values aren't the same OR no value for that key in old metadata."""
         try:
             old_val = _get_val_in_metadata_dotted(field, old_metadata)
             return bool(val != old_val)
         except utils.DottedKeyError:
-            return True
+            return True  # value was not found in old_metadata
 
-    def has_forbidden_attributes_creation(self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata) -> bool:
-        """Check if `metadata` has forbidden attributes and they have changed.
-
-        Returns `True` if it has forbidden attributes.
-        """
-        forbidden_matches = self._find_all_field_vals(metadata, self.FORBIDDEN_FIELDS_CREATION)
-
-        for field, val in forbidden_matches.items():
-            if self._field_vals_are_different(field, val, old_metadata):
-                apihandler.send_error(
-                    400,
-                    reason=f"Validation Error: forbidden attribute creation '{field}'",
-                    file=apihandler.files_url,
-                )
-                return True
-        return False
-
-    def has_forbidden_attributes_modification(self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata) -> bool:
-        """Check if `metadata` has forbidden attribute updates."""
-        forbidden_matches = self._find_all_field_vals(metadata, self.FORBIDDEN_FIELDS_UPDATE)
+    @staticmethod
+    def _has_forbidden_attributes(
+        apihandler: Any,
+        metadata: types.Metadata,
+        old_metadata: types.Metadata,
+        forbidden_attributes: List[str],
+        http_error_message: str,
+    ) -> bool:
+        forbidden_matches = Validation._find_all_field_vals(
+            metadata, forbidden_attributes
+        )
 
         for field, val in forbidden_matches.items():
-            if self._field_vals_are_different(field, val, old_metadata):
+            if Validation._field_vals_are_different(field, val, old_metadata):
                 apihandler.send_error(
                     400,
-                    reason=f"Validation Error: forbidden attribute update '{field}'",
+                    reason=f"Validation Error: {http_error_message} '{field}'",
                     file=apihandler.files_url,
                 )
                 return True
         return False
 
-    def validate_metadata_creation(self, apihandler: Any, metadata: types.Metadata) -> bool:
-        """Validate metadata for creation.
+    def has_forbidden_attributes_creation(
+        self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata
```

Comment:
I think this signature is consistent with `has_forbidden_attributes_modification` below, but does it need `old_metadata` in the case of creation?

On line 480 below, you are passing in `{}`, as an empty object to compare to (which makes sense, in the case of creation).

I think you might be able to remove `old_metadata` as a parameter and just define it as a local variable in the function like: `old_metadata = {}` between line 130 and 131.

Reply:
I'm glad you caught this. I'd been wondering about this myself, most likely it was an artifact of some refactoring.